File: react-app/src/services/controllers/user-controller.js
import Service from "../service";

export default class UserController extends Service {

    USER_URL = "user/";

    save = async (user) => {
        const response = await this.POST(this.USER_URL, user);
        console.log(response);
        return this.promise(response);
    };

    getByLogin = async (login) => {
        const response = await this.GET(this.USER_URL + `login=${login}`);
        console.log(response);
        return this.promise(response);
    };

    addResume = async (userId, resume) => {
        const response = await this.PUT(this.USER_URL + `user=${userId}`, resume);
        console.log(response);
        return this.promise(response);
    };

    deleteResume = async (userId, resume) => {
        const response = await this.PUT(this.USER_URL + `user=${userId}&resume=${resume}`, resume);
        console.log(response);
        return this.promise(response);
    };

    delete = async (userId) => {
        const response = await this.DELETE(this.USER_URL + `user=${userId}`);
        console.log(response);
        return this.promise(response);
    };
}

File: Java/src/main/java/RSL/controllers/UserController.java
package RSL.controllers;


import RSL.entities.*;
import RSL.models.EducationLevel;
import RSL.models.Gender;
import RSL.models.LanguageProficiencyLevel;
import RSL.models.RestResponse;
import RSL.services.ResumeService;
import RSL.services.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.Date;
import java.util.LinkedList;
import java.util.List;

@RestController
@RequestMapping("user")
@CrossOrigin(origins = "*", maxAge = 3600)
public class UserController {

    @Autowired
    private UserService userService;

    /**
     * Получить по логину
     * @param login
     * @return
     */
    @GetMapping
    public RestResponse<User> getByLogin(String login) {

        return new RestResponse<>(
                HttpStatus.OK,
                "getByLogin",
                userService.getByLogin(login)
        );
    }

    @PostMapping
    public RestResponse<User> save(User user) {
        return new RestResponse<>(
                HttpStatus.OK,
                "save",
                userService.save(user)
        );
    }

    /**
     * Удаление пользователя
     * @param user
     * @return
     */
    @DeleteMapping("(id)")
    public RestResponse<User> delete (@PathVariable("id") User user) {
        return new RestResponse<>(
                HttpStatus.OK,
                "delete",
                userService.delete(user)
        );
    }

    /**
     * <----------------------РЕЗЮМЕ ПОЛЬЗОВАТЕЛЯ------------------------->
     */

    /**
     * Добавление резюме пользователю.
     * @param user
     * @param resume
     * @return
     */
    @PutMapping("user={id}")
    public RestResponse<Resume> addResume(@PathVariable("id") User user, @RequestBody Resume resume) {
        return new RestResponse<>(
                HttpStatus.OK,
                "OKOK",
                userService.addResume(user, resume)
        );
    }

    @DeleteMapping("user={user_id}&resume={resume_id}")
    public RestResponse<Resume> deleteResume(@PathVariable("user_id") User user, @PathVariable("resume_id") Resume resume) {
        return new RestResponse<>(
                HttpStatus.OK,
                "deleteResume",
                userService.deleteResume(user, resume)
        );
    }




}

